documentation/VSTestIntegration.md

```diff
@@ -0,0 +1,92 @@
+
+#### Default
+| Option | Summary |
+|-------------|------------------------------------|
+|Format              | Results format in which coverage output is generated. Default format is cobertura.| 
+
+#### Advanced Options (Supported via runsettings)
+These are a list of options that are supported by coverlet. These can be specified as datacollector configurations in the runsettings.
+
+| Option         | Summary                                                                                  |
+|-------------   |------------------------------------------------------------------------------------------|
+|Format          | Coverage output format. These are either cobertura, json, lcov, opencover or teamcity.   | 
+|MergeWith       | Combine the output of multiple coverage runs into a single result.                       | 
+|Exclude         | Exclude from code coverage analysing using filter expressions.                           | 
+|ExcludeByFile   | Ignore specific source files from code coverage.                                         | 
+|Include         | Explicitly set what to include in code coverage analysis using filter expressions.       | 
+|IncludeDirectory| Explicitly set which directories to include in code coverage analysis.                   |
+|SingleHit       | Specifies whether to limit code coverage hit reporting to a single hit for each location.| 
+|UseSourceLink   | Specifies whether to use SourceLink URIs in place of file system paths.                  |
+
+How to specify these options via runsettings?
+```
+<?xml version="1.0" encoding="utf-8" ?>
+<RunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="XPlat code coverage">
+        <Configuration>
+          <Format>json</Format>
+          <MergeWith>/custom/path/result.json</MergeWith>
+          <Exclude>[coverlet.*.tests?]*,[*]Coverlet.Core*</Exclude> <!-- [Assembly-Filter]Type-Filter -->
+          <Include>[coverlet.*]*,[*]Coverlet.Core*</Include> <!-- [Assembly-Filter]Type-Filter -->
+          <ExcludeByAttribute>Obsolete,GeneratedCodeAttribute,CompilerGeneratedAttribute</ExcludeByAttribute>
+          <ExcludeByFile>../dir1/class1.cs,../dir2/*.cs,../dir3/**/*.cs,</ExcludeByFile> <!-- Absolute or relative file paths -->
+          <IncludeDirectory>../dir1/*,../dir2/,</IncludeDirectory>
+          <SingleHit>false</SingleHit>
+          <UseSourceLink>true</UseSourceLink>
+        </Configuration>
+      </DataCollector>
+    </DataCollectors>
+  </DataCollectionRunSettings>
+</RunSettings>
+```
+This runsettings file can easily be provided using command line option as given :
+
+1. `dotnet test --settings coverletArgs.runsettings`
+
+2. `dotnet vstest --settings coverletArgs.runsettings`
+
+
+#### Scope of Enhancement 
+Currently, advanced options are supported via runsettings. Providing support through additional command line arguments in vstest can be taken up separately.
```

Comment:
> Providing support through additional command line arguments in vstest can be taken up separately.

Will be better user experience...but I think that it's require non trivial internal changes.

Reply:
Yes. This will include some internal vstest changes which we can pick up as an enhancement.